### DISCUSSION TITLE
Vote for the next feature

### DISCUSSION THREAD

**Discussion by TrianguloY** [TrianguloY/URLCheck#383] [Polls]
This poll is now closed (but you can't close a github poll...)

---

This is some sort of experiment, let's see how it goes.

I know I've been a bit absent lately (more than usual), but I've kept working on the app from time to time. In particular I have prepared 3 features that are almost ready (the hard part is done) and only a final cleanup, bug checks and minor improvements are left.
So, I had the idea to perform some sort of feature vote.
Before you panic, no, I'm not going to let you choose only one. I'll add all three eventually, this vote is going to decide the order in which I'll add them. Maybe I'll be able to add two features on the same version, but most probably they'll be on different ones (and knowing myself, it may take a while between them).

To vote for each feature, just mark the option below, and/or leave a comment. In a week approx I'll count the votes (by 12 October 2024).

Also, I'm going to make the vote of those that donated count more, depending on the donation, specially (but not exclusive to) the places where that's a tier perk (I think this is a good opportunity for that). I'll make sure to check if all the voters have donated or not some way or another, ~but feel free to~ leave a comment or send me an email ~if you prefer to be sure~.
IMPORTANT! Turns out that Github polls don't show who voted what, and I didn't noticed sooner I'm sorry. If you have donated please tell me, otherwise I won't be able to know it :(

The three features are:

---

<details>
<summary>Virus Total V3</summary>
From Issue #87. This improvement migrates the VirusTotal Api from the v2 to the v3 version. This allows to retrieve more information about urls, plus I'll update the UI to better show the new information.

![vt](https://github.com/user-attachments/assets/dd3c2f09-0391-4c98-8f14-e7b34de7fad9)

Note: the screenshot shows a very basic secondary dialog, I'll probably add the information in the main one (and definitely make the link clickable).
</details>

---

<details>
<summary>Full activity Json Editor</summary>
The current json editor is shown as a dialog, which feel constrained, specially with big jsons. I do want to add a better editor for some features, but since the json editor will always be there, this is probably a nice improvement. (Maybe in the far future I'll add a better editor with line numbers and syntax highlight and auto-completion? Unlikely, but this is a very small step towards that)

![json](https://github.com/user-attachments/assets/9930db4b-17e3-41df-9607-926c164acf99)

</details>

---

<details>
<summary>Automation</summary>
From issues #18, #21, #192 and #279 (and probably a couple more I forgot).
The Status code module has an 'automatic' feature that will basically press the button for urls that match a given pattern. The idea is to extract that feature to allow all modules to automatically perform actions based on the url.
This can become a very big feature, with a lot of possibilities, but for now it will just be a simple 'if url matches <regexp> then press <button>'. I do want to add the ability to use other inputs (like modules outputs, or app properties like the referrer) but we'll see about that.

![auto](https://github.com/user-attachments/assets/0c008f80-bcb0-4790-a7ca-69a2d3819a2a)

</details>

---

Will this be a somewhat moderately interesting experiment, or an absolute failure? We'll see!

**Comment by TrianguloY** [comment]
IMPORTANT! Turns out that Github polls don't show who voted what, and I didn't noticed sooner, I'm sorry. If you have donated please tell me, otherwise I won't be able to know it :(

**Reply by vs4vijay** [reply]
+1 for automations

**Comment by TrianguloY** [comment]
Ok! Time's up.

In the end I got 7 votes. From those:
- 4 votes for Automation. The winner. This will be available on the next version, hopefully this month (but can't promise anything).
- 2 votes for the Json full activity. This will be added next. Since it will help a lot with the automation I will try to release it soon afterwards, but this is the most tricky feature so it may take a bit more.
- 1 vote for the Virus Total V3. The last one, which I guess it's useful but less than the others (the current v2 already works fine).

There was one possible extra vote for the automation from a donation supporter (?) ... I'm sorry but I couldn't find you with that username in any of the donation platforms I have. In any case it doesn't change the result.


As an experiment, this poll wasn't a big deal, but wasn't a total failure either, so I'm still happy. But if I ever do something similar again, I won't use Github polls (unless they greatly improve them) because as it is now they're very VERY limited (you can't see who voted, you can't set it for a limited time and you can't even close it, these features have been suggested already to github btw).

Thanks for those who participated!

**Reply by Lppsoeht** [reply]
Hello, loving the new feature! Perhaps it would be better if someone would share their json file so others, also not tech savy, would benefit from a larger urlcheck automation.

**Reply by TrianguloY** [reply]
Thank you! 
I personally only use the default bit.ly one (that's why it's the default ;) but I'll be happy to add others here to the wiki. Or even if it's a useful and interesting one I can add it to the app as default too!
